### PR TITLE
Add ability to toggle page titles on and off

### PR DESCRIPTION
**Pull request by AnthonyLedesma**
goHidePageTitle

- [x] Persist data in post meta.
- [x] Hide titles when setting is active.
- [x] Default setting based on the customizer selection.
  * The default setting used in Customizer is now used when creating new pages.

### CODE REVIEW

File: includes/title-meta.php [hunk 1]
```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * Page Title Meta setup
+ *
+ * @package Go\TitleMeta
+ */
+
+namespace Go\TitleMeta;
```
Comment (EvanHerman) [inline includes/title-meta.php:8]: @AnthonyLedesma Lets update this class name to `Go\Title_Meta` just so we're matching the style in our other classes (`Title_Control`, `Switcher_Control`, `Range_Control` etc.). Also, I'm fairly certain core uses the same naming conventions.

File: includes/title-meta.php [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+function page_title_add_meta_boxes( $post ) {
+	add_meta_box(
+		'page_title_checkbox',
+		__( 'Hide page title', 'go' ),
+		'Go\TitleMeta\\page_title_build_metabox',
+		'page',
+		'side',
+		'high'
+	);
+
```
Comment (EvanHerman) [inline includes/title-meta.php:55]: You can remove the space here so all methods don't have an empty space at the beginning and end of them.

File: includes/title-meta.php [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+ */
+function page_title_build_metabox( $post ) {
+	$label = __( 'Hide page title when published.', 'go' );
+	wp_nonce_field( basename( __FILE__ ), 'page_title_metabox_nonce' );
+
+	?>
+	<div class='page-title-meta'>
+		<p>
+			<input type="checkbox" id="page-title-checkbox" name="page-title" value="on" <?php checked( get_post_meta( $post->ID, '_page_title', true ), 1 ); ?> />
+			<label for="page-title-checkbox"> <?php echo( esc_html( $label ) ); ?> </label><br>
```
Comment (EvanHerman) [inline includes/title-meta.php:71]: Remove the space before and after the `$label`.

```php
<label for="page-title-checkbox"><?php echo( esc_html( $label ) ); ?></label><br>
```

Comment (EvanHerman) [inline includes/title-meta.php:71]: I guess while we're at it, we can unify the `$label` on line 64 and where we output it on line 71.

```php
<label for="page-title-checkbox"><?php esc_html_e( 'Hide page title when published.', 'go' ); ?></label><br>
```

File: includes/title-meta.php [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+ * @param int $post_id The post ID.
+ * @link https://codex.wordpress.org/Plugin_API/Action_Reference/save_post
+ */
+function page_title_meta_box_data( $post_id ) {
+
+	$page_title_metabox_nonce = filter_input( INPUT_POST, 'page_title_metabox_nonce', FILTER_SANITIZE_STRING );
+
+	if ( ! $page_title_metabox_nonce || ! wp_verify_nonce( $page_title_metabox_nonce, basename( __FILE__ ) ) ) {
+
+		return;
```
Comment (EvanHerman) [inline includes/title-meta.php:91]: We may as well unify all of these returns, so we've just got a single return in this method. This should work:

```php
if (
	! current_user_can( 'edit_post', $post_id ) ||
	( defined( 'DOING_AUTOSAVE' ) && DOING_AUTOSAVE ) ||
	! $page_title_metabox_nonce ||
	! wp_verify_nonce( $page_title_metabox_nonce, basename( __FILE__ ) ) ||
) {

	return;

}
```

File: includes/title-meta.php [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+/**
+ * Hide page title box
+ *
+ * @param title_data $title_data The title data.
+ * @return title_data $title_data Filtered title data.
+ */
+function hide_page_title( $title_data ) {
+	if ( ! is_singular( 'page' ) || ! get_post_meta( get_the_ID(), '_page_title', true ) ) {
+		return $title_data;
+	} else {
```
Comment (EvanHerman) [inline includes/title-meta.php:34]: We should try and avoid using `if/else` statements where possible. We can actually just set`$title_data['title'] = '';` when `is_singluar( 'page' ) && get_post_meta( get_the_ID(), '_page_title', true )`:

```php
/**
 * Hide page title box
 *
 * @param title_data $title_data The title data.
 * @return title_data $title_data Filtered title data.
 */
function hide_page_title( $title_data ) {
	if ( is_singular( 'page' ) && get_post_meta( get_the_ID(), '_page_title', true ) ) {
		$title_data['title'] = '';
	}
	return $title_data;
}
```

**Comment by fjarrett** [comment]
@richtabor Is there another top-level sidebar section where this option could live rather than adding a new one?

**Comment by richtabor** [comment]
@EvanHerman @AnthonyLedesma Is there a way to add this to the core Page Attributes panel? 
Screen Shot 2020-04-29 at 12 41 44 PM

**Comment by EvanHerman** [comment]
@richtabor I don't think so but we can take a second look 👀

**Comment by AnthonyLedesma** [comment]
@richtabor 
I cannot see any supported way to accomplish this. We could get around it with JavaScript but it'd be hacky.

**Comment by richtabor** [comment]
@AnthonyLedesma What about in Status & Visibility? Screen Shot 2020-04-29 at 1 07 48 PM

**Comment by AnthonyLedesma** [comment]
@richtabor Looks like Gutenberg supports extending this control. 
https://github.com/WordPress/gutenberg/blob/3dc8ebb8c933e7d1095863994b2a1f375c98c0ff/packages/edit-post/src/components/sidebar/plugin-post-status-info/index.js

There is a bit of cross over where we use the PHP API for the metabox but this sidebar location uses a JavaScript API. Possible but will require considerable effort for the refactor.

**Pull request merged**